src/net/rpc.rs

```diff
@@ -76,3 +92,37 @@ async fn handle_socket(ws: WebSocket, sess: Session, id: Uuid) {
 	// Serve the socket connection requests
 	Connection::serve(rpc, ws).await;
 }
+
+async fn post_handler(
+	Extension(session): Extension<Session>,
+	output: Option<TypedHeader<Accept>>,
+	content_type: TypedHeader<ContentType>,
+	body: Bytes,
+) -> Result<impl IntoResponse, impl IntoResponse> {
+	let fmt: Format = content_type.deref().into();
+	let out_fmt: Option<Format> = output.as_deref().map(Into::into);
+	if let Some(out_fmt) = out_fmt {
+		if fmt != out_fmt {
+			return Err(Error::InvalidType);
+		}
+	}
```

Comment:
Do we really need this check? It's probably not a good pattern, but it somebody wants a different output format for some reason I don't see a reason to disallow it